### PR TITLE
Added a category to templates

### PR DESCRIPTION
**Pull request by jnolis**
Our list of templates is getting long and difficult for users to parse. As part of @dotNomad's work in redoing the "create from template" flow, I think it would be a good idea to have the template broken down into categories, like:

* Production (jobs, apis, dashboards)
* Deep learning (PyTorch, Tensorflow)
* Parallel (dask, future)

And so on.

I also considered adding a separate "language" for further filtering, but I think that's overkill for now. I also made this an optional parameter, so like for enterprise customers they don't have to categorize if they don't want to.

### CODE REVIEW

File: templates/schema.json [hunk 1]
```diff
@@ -11,6 +11,10 @@
       "type": "string",
       "description": "Longer description of the template."
     },
+    "category": {
```
Comment (jsignell) [inline templates/schema.json:14]: Alternate idea from github-land: "labels"

File: templates/schema.json [hunk 2]
```diff
@@ -11,6 +11,10 @@
...
+      "type": "string",
+      "description": "Optional category to use when showing a list of templates."
```
Comment (jsignell) [inline templates/schema.json:16]: I've been trying to set [FUTURE RELEASE] on these until there is anything implemented in Saturn to support.

**Review by jsignell — COMMENTED** [review]
For the language thing, I think there is a lot of overlap with `supports` on images but I also think that will be kind of repetetive because we already know what type of resource the template creates (we can just look in the resource recipe and see which subdicts exist (jupyter_server, dask_cluster, rstudio...)

File: templates/schema.json (hunk 1 above)
Comment (jnolis) [reply templates/schema.json:14]: to me labels suggests an array not a string, which I am open to but would add some other complications to my original implementation of having things in at most one bucket on the list.

Comment (jsignell) [reply templates/schema.json:14]: Oh yeah definitely labels would be an array.

Comment (jnolis) [reply templates/schema.json]: okay i switched from a category string to a label array.

Once we merge it you can start using it when thinking through the template list view UI Jordan

**Review by jsignell — APPROVED** [review]
Nice!

**Review by dotNomad — APPROVED** [review]
Nice and clean :sunglasses:
